lib/shopping_cart.rb

```diff
@@ -0,0 +1,56 @@
+class ShoppingCart
+  attr_reader :name,
+              :products
+
+  def initialize(name, capacity)
+    @name = name
+    @capacity = capacity
+    @products = []
+  end
+
+  def capacity
+    @capacity.to_i
+  end
+
+  def add_product(product)
+    products << product
+  end
+
+  def details
+    details_hash = {name: name, capacity: capacity}
+  end
+
+  def total_number_of_products
+    products.reduce(0) do |sum, product|
+      sum += product.quantity
+    end
+  end
+
+  def is_full?
+    return true if total_number_of_products > capacity
+    false
+  end
```

Comment:
You could shorten up this method by relying on the return value of the condition to produce either true or false (instead of the if logic)